src/benchmark/trulens/benchmark/benchmark_frameworks/experiments/competitive_analysis.ipynb

```diff
@@ -0,0 +1,573 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #68.        write_results(
</pre><p>If we are initializing a trulens app, why write results? Also we are not using tru benchmark experiment? Was that not supposed to help here?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/truera/trulens/pull/1559/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Reply:
<p>these utils (along with a lot of other stuff) were from the time when I first started running benchmark back in September, and during those periods of time TruBenchmark wrapper was not in a good enough shape to be used. </p><p> </p><p>"not in a good enough shape" here refers to 2 main thing</p><p> </p><ol><li>score generation errors throw off the execution of TruBenchmark app, cuz all scores are generated from one app run </li><li>latency display and casting issue, which was later fixed by Corey's PR <a href="https://github.com/truera/trulens/pull/1522" target="_blank">https://github.com/truera/trulens/pull/1522</a></li></ol>

Reply:
<p>I'll have a separate PR to refactor. This PR is mainly just moving things around  from existing RAG triad benchmark notebook to <code>dataset_preprocessing.py</code></p>